test/functional-portable/corerp/noncloud/resources/recipe_bicep_test.go

```diff
@@ -392,13 +392,26 @@ func Test_BicepRecipe_LanguageFailure(t *testing.T) {
 				Code:            recipes.RecipeDeploymentFailed,
 				MessageContains: "failed to deploy recipe",
 				Details: []step.DeploymentErrorDetail{
+					// The column index flagged for the error is different depending on what version of the extension is used.
+					// These values might be different between functional test and long running test runs.
+					// Instead of hardcoding values, we'll validate that the rest of the message is correct.
 					{
 						Code:            "DeploymentFailed",
 						MessageContains: "At least one resource deployment operation failed. Please see the details for the specific operation that failed.",
 						Details: []step.DeploymentErrorDetail{
 							{
 								Code:            "InvalidTemplate",
-								MessageContains: "Unable to process template language expressions for resource 'Applications.Core/extenders/corerp-resources-recipe-bicep-langugagefailure-failure' at line '1' and column '456'. 'Unable to evaluate the template language function 'substring'. The index parameter cannot be larger than the length of the string. The index parameter: '10', the length of the string parameter: '4'. Please see https://aka.ms/arm-function-substring for usage details.'",
+								MessageContains: "Unable to process template language expressions for resource 'Applications.Core/extenders/corerp-resources-recipe-bicep-langugagefailure-failure' at line '1' and column",
```

Comment:
Depending on error message matching is generally not reliable since they can change at any point, service can't change error code but there is no such contract for message strings. Let's consider removing this, if this becomes a pain point. We already have error code validation, so I'm not sure if this really adds extra value.